### PR TITLE
Ceds 2036 Split departure transport + skip departure transport on clearance request

### PR DESCRIPTION
**Pull request by bambuchaAdm**

### CODE REVIEW

File: app/connectors/exchange/ExportsDeclarationExchange.scala [hunk 1]
```diff
@@ -94,7 +94,7 @@ object ExportsDeclarationExchange {
     dispatchLocation: Option[DispatchLocation] = None,
     additionalDeclarationType: Option[AdditionalDeclarationType] = None,
     consignmentReferences: Option[ConsignmentReferences] = None,
-    departureTransport: Option[DepartureTransport] = None,
+    departureTransportOld: Option[(Option[String], Option[DepartureTransport])] = None,
```
Comment (ed0906) [inline app/connectors/exchange/ExportsDeclarationExchange.scala]: Why?

We haven't had to do this for any other of these "split" stories?

File: app/connectors/exchange/ExportsDeclarationExchange.scala [hunk 2]
```diff
@@ -183,7 +186,9 @@ object ExportsDeclarationExchange {
       (__ \ "dispatchLocation").readNullable[DispatchLocation] and
       (__ \ "additionalDeclarationType").readNullable[AdditionalDeclarationType] and
       (__ \ "consignmentReferences").readNullable[ConsignmentReferences] and
-      (__ \ "departureTransport").readNullable[DepartureTransport] and
+      (__ \ "departureTransport").readNullable({
```
Comment (ed0906) [inline app/connectors/exchange/ExportsDeclarationExchange.scala]: Why do we still have 2 reads?

I thought we agreed this wasn't necessary?

**Comment by ed0906** [comment]
I think this task has taken a lot longer than it should have because of scope creep :(
In my opinion, the problem of versioning should been a technical ticket (and solved as one)

File: app/connectors/exchange/ExportsDeclarationExchange.scala (hunk 1 above)
Comment (bambuchaAdm) [reply app/connectors/exchange/ExportsDeclarationExchange.scala]: No backend speak with version1 but understood version1 and version2. Now frontend could undestood version1 and version2 and writes to backend with version2.

**Comment by bambuchaAdm** [comment]
@PatrykRudnicki After this PR we are able to remove second reads from backend and bump writes to version2 - after that PR we would be able to remove second reads end with only one.

@ed0906 It was little widen though work made by @PatrykRudnicki on summary page - required more place to update `ModeOfTransportCode`

**Pull request merged**